watertap/unit_models/pressure_changer.py

```diff
@@ -11,35 +11,301 @@
+                    b.flow_ratio[t] < 0.6,
+                    0.4,
+                    Expr_if(
+                        b.flow_ratio[t] > 1.4,
+                        0.4,
+                        -0.995 * b.flow_ratio[t] ** 2 + 1.977 * b.flow_ratio[t] + 0.018,
+                    ),
+                )
+
+        elif self.config.variable_efficiency is VariableEfficiency.flow_head:
+            raise ValueError(
+                "Config option 'VariableEfficiency.flow_head' is not fully implemented yet"
+            )
+            # TODO - Implement pump efficiency expression based on flow and head
+            # self.bep_head = Var(
+            #     initialize=1.0,
+            #     doc="Best efficiency point head of the centrifugal pump",
+            #     units=pyunits.m,
+            # )
+            #
+            # self.head_ratio = Var(
+            #     self.flowsheet().time,
+            #     initialize=1.0,
+            #     doc="Ratio of pump head to best efficiency point head",
+            #     units=pyunits.dimensionless,
+            # )
+            #
+            # @self.Constraint(self.flowsheet().time, doc="Pump head ratio")
+            # def head_ratio_constraint(b, t):
+            #     if b.control_volume.properties_in[t].dens_mass_phase["Liq"] is None:
+            #         raise Exception(
+            #             "Density is not implemented in property package as 'dens_mass_phase'"
+            #         )
+            #
+            #     return b.head_ratio[t] * b.bep_head * Constants.acceleration_gravity == (
+            #             (
+            #                     b.control_volume.properties_out[t].pressure
+            #                     / b.control_volume.properties_out[t].dens_mass_phase["Liq"]
+            #             )
+            #             - (
+            #                     b.control_volume.properties_in[t].pressure
+            #                     / b.control_volume.properties_in[t].dens_mass_phase["Liq"]
+            #             )
+            #     )
+        if self.config.variable_efficiency is not VariableEfficiency.none:
+            # replace the constant efficiency assumption using eta_ratio
+            # must be done after the eta_ratio expression is created.
+            @self.Constraint(self.flowsheet().time, doc="Actual pump efficiency")
+            def eta_constraint(b, t):
+                return b.efficiency_pump[t] == (b.bep_eta * b.eta_ratio[t])
+
+    def calculate_scaling_factors(self):
+        super().calculate_scaling_factors()
+
+        for ind, c in self.control_volume.isothermal_balance.items():
+            sf = iscale.get_scaling_factor(
+                self.control_volume.properties_in[0].temperature
+            )
+            iscale.constraint_scaling_transform(c, sf)
+
+        if hasattr(self, "bep_flow"):
+            if iscale.get_scaling_factor(self.bep_flow) is None:
+                iscale.set_scaling_factor(self.bep_flow, 10)
```

Comment:
This is likely one the user should provide, so we can have a default scaling factor with a warning. I would opt to set it to the inverse of whatever bep_flow value is (or something close to that).

Reply:
@arao53 I think this suggested change still stands, but it might not be much of an issue. I just thought that anything related to flowrate will highly depend on the user's decision, so perhaps there should be a warning to the user indicating such.

Reply:
@adam-a-a sorry about that! Somehow this got lost along the way. I'll update that shortly.